test/e2e/basic_test.go

```diff
@@ -654,6 +654,90 @@ var _ = Describe("Third party objects", func() {
 		}, 30*time.Second, 1*time.Second).Should(BeNil())
 	})
 
+	It("Daemon set version cleans orphaned IP claims when scheduler has missed services updates for some reason", func() {
+		processName := "ipmanager"
+		By("deploying claim scheduler pod")
+		scheduler := newPod(
+			"externalipscheduler", "externalipcontroller", "mirantis/k8s-externalipcontroller",
+			[]string{processName, "s", "--mask=24", "--logtostderr", "--v=5"}, nil, false, false)
+		_, err := clientset.Core().Pods(ns.Name).Create(scheduler)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("deploying claim controller daemon set")
+		// sh -c will be PID 1 and we will be able to stop our process
+		cmd := []string{processName, "c", "--logtostderr", "--v=5", "--iface=docker0"}
+		ds := newDaemonSet("externalipcontroller", "externalipcontroller", "mirantis/k8s-externalipcontroller",
+			[]string{"sh", "-c", strings.Join(cmd, " ")}, ipcontrollerLabels, true, true)
+		ds, err = clientset.Extensions().DaemonSets(ns.Name).Create(ds)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("waiting until both nodes will be registered")
+		Eventually(func() error {
+			ipnodes, err := ext.IPNodes().List(api.ListOptions{})
+			if err != nil {
+				return err
+			}
+			if len(ipnodes.Items) != 2 {
+				return fmt.Errorf("Unexpected nodes length %v", ipnodes.Items)
+			}
+			return nil
+		}, time.Second * 30, 2 * time.Second).Should(BeNil())
+
+		By("deploying nginx pod and service with multiple external ips")
+		nginx1Name := "nginx1"
+		var nginx1Port int32 = 2288
+		_, network, _ = net.ParseCIDR("10.107.10.0/24")
+		Expect(err).NotTo(HaveOccurred())
+		externalIPs1 := []string{"10.107.10.2", "10.107.10.3"}
+		deployNginxPodAndService(nginx1Name, nginx1Port, clientset, ns, externalIPs1)
+
+		By("deploying second nginx pod and service with multiple external ips")
+		nginx2Name := "nginx2"
+		var nginx2Port int32 = 2289
+		externalIPs2 := []string{"10.107.10.3", "10.107.10.4"}
+		deployNginxPodAndService(nginx2Name, nginx2Port, clientset, ns, externalIPs2)
+
+		By("assigning ip from external ip pool to a node where test is running")
+		Expect(netutils.EnsureIPAssigned(testutils.GetTestLink(), "10.107.10.10/24")).Should(BeNil())
+
+		By("verifying that both nginx services are reachable")
+		verifyServiceReachable(nginx1Port, externalIPs1...)
+		verifyServiceReachable(nginx2Port, externalIPs2...)
+
+		By("checking quantity of allocated ip claims")
+		claims := getAllocatedClaims(ext)
+		Expect(len(claims)).To(BeNumerically("==", 3))
+
+		By("shuting down scheduler")
+		var zero int64 = 0
+		err = clientset.Core().Pods(ns.Name).Delete(scheduler.Name, &api.DeleteOptions{
+			GracePeriodSeconds: &zero,
+		})
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
i think you need to wait until scheduler will be removed and then check that claims == 3, otherwise we wont know why claim was removed